source/exe/hot_restart.h

```diff
@@ -138,8 +138,12 @@ class HotRestartImpl : public HotRestart,
   };
 
   struct RpcBase {
-    RpcBase(RpcMessageType type, uint64_t length = sizeof(RpcBase))
-        : type_(type), length_(length) {}
+    RpcBase(RpcMessageType type, uint64_t length = sizeof(RpcBase)) {
+      static_assert(std::is_standard_layout<RpcBase>::value, "is_standard_layout<RpcBase>");
```

Comment:
After thinking about this a bit, I think we have a deeper problem with using RPC structs with padding. Different compilers in principle might apply different amounts of padding (unlikely, but possible). So, the new Envoy instance might be built with clang while the old one with gcc for example. We should really be using structs with no padding, or even better some proper serialization format like protobuf.

I'm borderline about how likely this is in reality to happen on a given architecture (which is constant across upgrades) and with very simple POD, which have fairly well defined and understood alignment requirements.

How disruptive is a version bump to clean this up? If you think it's not worth fixing since most folks will build with a consistent compiler, we can do one of the above solutions.